dvc/remote/gdrive.py

```diff
@@ -73,6 +74,17 @@ def __init__(self, repo, config):
         super(RemoteGDrive, self).__init__(repo, config)
         self.no_traverse = False
         self.path_info = self.path_cls(config[Config.SECTION_REMOTE_URL])
+        bucket_case_sensitive = re.search(
```

Comment:
I'm not sure what's going on here? Could you elaborate please? Should we put a comment?

Reply:
further gdrive logic requires top folder id ("bucket") from URL. For example, for gdrive://0AIac4JZqHhKmUk9PDA/test , folder "test" will be created/used from inside folder with id `0AIac4JZqHhKmUk9PDA`. "bucket" can be "root" or unique folder id.

`self.path_info` already contains parsed `config[Config.SECTION_REMOTE_URL]`, but `urlparse` converts whole URL to lower case.

This source code statement searches for already parsed and lower cased "bucket" part of URL with `IGNORECASE` option in original `Config.SECTION_REMOTE_URL` and retrieves original not lower cased "part" of URL.

For example, it searches for `0aiac4jzqhhkmuk9pda` in `gdrive://0AIac4JZqHhKmUk9PDA/test ` and returns `0AIac4JZqHhKmUk9PDA`.

Reply:
@MaxRis Maybe it is easier to just parse the url? :)

Reply:
@MaxRis Also, is `0AIac4JZqHhKmUk9PDA` a team drive id? Is that notation compatible with regular non-team drive urls?

Reply:
@efiop 

> @MaxRis Maybe it is easier to just parse the url? :)
Will try with `re`

>  Also, is 0AIac4JZqHhKmUk9PDA a team drive id? Is that notation compatible with regular non-team drive urls?

It is the same unique id which has any Gdrive's file or folder. PyDrive internally adds additional API query params to extend query access for shared drives, but for user it is transparent and no difference if file/folder was accessed from usual or shared drive.

Reply:
@efiop have updated that

Reply:
https://github.com/iterative/dvc/pull/3016/commits/b4a663a981a182613f36ba3623ec845155ee7660#diff-21bfd10d01c6715fc8634644730fa720R82

I believe it shoudn't fail under any input with index error or regular expression group not found error.